services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+        private Integer maxInflightReceiveBatches;
+        private List<MessageSystemAttributeName> messageSystemAttributeNames = Collections.emptyList();
+        private List<String> receiveMessageAttributeNames = Collections.emptyList();
+        private Boolean adaptivePrefetching;
+
+        private Builder() {
+        }
+
+        /**
+         * Define the maximum number of messages that are batched together in a single request.
+         *
+         * @param maxBatchItems The new maxBatchItems value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchItems(Integer maxBatchItems) {
+            this.maxBatchItems = maxBatchItems;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of batchKeys to keep track of. A batchKey determines which requests are batched together and
+         * is calculated by the client based on the information in a request.
+         * <p>
+         * Ex. SQS determines a batchKey based on a request's queueUrl in combination with its overrideConfiguration, so requests
+         * with the same queueUrl and overrideConfiguration will have the same batchKey and be batched together.
+         *
+         * @param maxBatchKeys the new maxBatchKeys value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchKeys(Integer maxBatchKeys) {
+            this.maxBatchKeys = maxBatchKeys;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of items to allow to be buffered for each batchKey.
+         *
+         * @param maxBufferSize the new maxBufferSize value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBufferSize(Integer maxBufferSize) {
+            this.maxBufferSize = maxBufferSize;
+            return this;
+        }
+
+        /**
+         * Define the maximum amount of time that an outgoing call waits for other requests before sending out a
+         * batch request.
+         * TODO : Decide if Ms needs to be added to the name in surface API review meeting
+         * @param maxBatchOpenDuration The new maxBatchOpenDuration value.
+         * @return This object for method chaining.
+         */
+        public Builder maxBatchOpenDuration(Duration maxBatchOpenDuration) {
+            this.maxBatchOpenDuration = maxBatchOpenDuration;
+            return this;
+        }
+
+        /**
+         * Define the custom visibility timeout to use when retrieving messages from SQS. If set to a value greater than zero,
+         * this timeout will override the default visibility timeout set on the SQS queue. Set it to -1 to use the default
+         * visibility timeout of the queue. Visibility timeout of 0 seconds is not supported.
+         *
+         * @param visibilityTimeout The new visibilityTimeout value.
+         * @return This object for method chaining.
+         */
```

Comment:
How do I set a Duration to -1? -1 what? Generally using -1 or 0 to signify something other than -1 or 0 is a bad design. Should we use the default visibility timeout of the queue if this is null?

Why is a visibility timeout of 0 not supported?

What's the "resolution" of this setting? Can I do 1 millisecond? 1 nanosecond?